src/events/guild/messageCreate.ts

```diff
@@ -0,0 +1,119 @@
+import { Message } from "discord.js"
+import FAGCBot from "../../base/fagcbot"
+import { afterJoinGuild, sendToGuild } from "../../utils/functions"
+
+export default async (client: FAGCBot, message: Message) => {
+	if (message.author.bot) return
+	if (!message.inGuild()) return
+	const prefix = client.env.BOTPREFIX
+	if (!message.content.startsWith(prefix)) return
+
+	const args = message.content.slice(prefix.length).split(/ +/g)
+	const cmd = args.shift()
+	if (!cmd) return message.reply("Invalid command!")
+
+	const command =
+		client.commands.get(cmd) ||
+		client.commands.get(client.aliases.get(cmd) || "")
+	if (!command) return message.reply("Invalid command!")
+
+	// TODO: add user to rate limit
+
+	// this temp var is required due to TS being TS
+	// fetch the guild config for the guild, or create one if it doesn't exist yet
+	let tmpGuildConfig = await client.fagc.communities.fetchGuildConfigMaster({
+		guildId: message.guild.id,
+	})
+	if (!tmpGuildConfig) {
+		tmpGuildConfig = await client.fagc.communities.createGuildConfig({
+			guildId: message.guild.id,
+		})
+		// this creates a guild config if it doesn't exist yet, so it runs after
+		afterJoinGuild(message.guild, client)
+	}
+	const guildConfig = tmpGuildConfig
+
+	// check if the command requires api key
+	if (command.requiresApikey && !guildConfig.apiKey)
+		return message.channel.send(
+			`${client.config.emotes.warn} API key must be set for use of this command`,
+		)
+
+	// if command doesnt require guild config (like help, ping etc), it can be ran
+	if (!command.requiresRoles) {
+		return await command.run({
+			message,
+			args,
+			client,
+			guildConfig,
+		})
```

Comment:
A try catch is still required here because there is no catch higher in the stack, so there will be no user output.